flax/core/scope.py

```diff
@@ -14,13 +14,52 @@
 
 """Flax functional core: Scopes.
 
-The goal of the Flax functional core is to provide a purely functional 
-abstraction that takes care of various types of bookkeeping such as parameter
-and RNG management. The main abstraction is Scope, which contains parameters
-and RNGs, and which can be nested.
-
-TODO: elaborate.
-
+Flax deep learning models contain variables, which can be of different 
+categories such as trainable parameters, or internal state variables that aren't
+parameters (e.g., batch statistics). For instance, the top-level variable
+dictionary of a Flax model using batch normalization may look as follows::
+  {
+    "params": {
+      "Conv1": { "weight": ..., "bias": ... },
+      "BatchNorm1": { "scale": ..., "mean": ... },
+      "Conv2": {...}
+    },
+    "batch_stats": {
+      "BatchNorm1": { "moving_mean": ..., "moving_average": ...}
+    }
+  }
+
+This variable dict has two so-called "collections": `params` (containing the 
+trainable parameters) and `batch_stats` (containing the --non-trainable-- batch
+statistics).
+
+The functional core Scope defined below allows easy access to all variables in a
+layer. For instance, from inside the BathcNorm1 layer, the variable "mean" is
+accessed as follows::
+  mean = scope.variable('batch_stats', 'moving_mean', jnp.zeros, (,))
+
+This will initialize it using `jnp.zeros` with shape `(,)`. The function 
```

Comment:
this shape here I find a bit confusing. `(,)` is the same as `()` which is a scalar. But batch stats are not scalars